### PR TITLE
Color Fixes

### PR DESCRIPTION
**Pull request by timkinnane**
@RocketChat/core 

Related (potentially fix) #2347 #1944 #2184 #3625 #3605 #4004

Refactored all use of color in less files. Includes some minor UI fixes.
Removed 99% of hard-coded hex/rgba values from style sheets - which was blocking any consistent use of admin color settings and leading a range of inconsistent UI styles.

My aim is to introduce a simpler color scheme with a clear hierarchy, not settings for every specific use. The current approach of naming colors for their components will continue bloating as new components are introduced. For developers who aren't proficient with design the semantics of the color scheme should help them make decisions, e.g. if you're adding a primary interaction, use the primary color. I'm trying to move towards the Google material design principles, where there's a consistent visual language for interactions, where it's clear from a glance what the main UI features are and you don't have components competing with each other visually or inventing mis-matching aesthetics. There's still lots of work to do. I will continue working on it, just putting the PR up to get some help and before I get further out of date with develop branch.

I've added an option to toggle minor colours for only those who really need them, otherwise they will fallback to major defaults. There's a problem here though, because the toggle `theme-option-extended-colors` setting has to be read at build to enable to the minor colour other settings to register. Would love someone to tell me a better way to do that.

I may have lost some styles because there's so much difference now its impossible to accurately resolve merge conflicts. Need help to review any recent additions or views I haven't tested (especially livechat).

Also need help with migrations. I had a go, but I'm not sure on the approach and I'm still getting old color settings appear in my admin pages, even after deleting them from mongo.

### CODE REVIEW

**Comment by lucasvanhalst** [comment]
Nice, this will make creating new themes so much easier

**Comment by timkinnane** [comment]
@lucasvanhalst yep, that's the idea.

File: packages/rocketchat-authorization/client/stylesheets/permissions.less [hunk 1]
```diff
@@ -4,78 +4,27 @@
...
-			td {
-				font-weight: bold;
-				a {
-					white-space: nowrap;
-				}
-				.icon-edit {
-					font-size: 80%;
-				}
-			}
+	    padding-left: 2px;
```
Comment (karlprieb) [inline packages/rocketchat-authorization/client/stylesheets/permissions.less]: Can you fix indentation here too? :)

**Comment by timkinnane** [comment]
Will update this due to #4803 and #4815 - I'm about to take changes by @karlprieb, fix some minor issues of my own and resubmit PR to the new branch.

**Pull request closed without merging**